File: JavaScript/167. Two Sum II - Input Array Is Sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    var a = numbers.length - 1, b = 0;
    while(numbers[a] + numbers[b] != target){
        if(numbers[a] + numbers[b] > target){ a--; }
        else{ b++; }
    }
    return [b + 1, a + 1];
};

File: C++/167. Two Sum II - Input Array Is Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int a = numbers.size() - 1, b = 0;
        while(numbers[a] + numbers[b] != target){
            if(numbers[a] + numbers[b] > target){ a--; }
            else{ b++; }
        }
        vector<int> result{b + 1, a + 1};
        return result;
    }
};
